server/routes/api/urls/urls.ts

```diff
@@ -0,0 +1,165 @@
+import { differenceInMinutes, formatDistanceToNowStrict } from "date-fns";
+import { t } from "i18next";
+import Router from "koa-router";
+import { head, orderBy } from "lodash";
+import { dateLocale } from "@shared/utils/date";
+import parseDocumentSlug from "@shared/utils/parseDocumentSlug";
+import parseMentionUrl from "@shared/utils/parseMentionUrl";
+import { NotFoundError } from "@server/errors";
+import auth from "@server/middlewares/authentication";
+import { transaction } from "@server/middlewares/transaction";
+import validate from "@server/middlewares/validate";
+import { Document, User } from "@server/models";
+import { authorize } from "@server/policies";
+import { APIContext } from "@server/types";
+import { opts } from "@server/utils/i18n";
+import * as T from "./schema";
+
+const router = new Router();
+
+router.post(
+  "urls.unfurl",
+  auth(),
+  validate(T.UrlsUnfurlSchema),
+  transaction(),
+  async (ctx: APIContext<T.UrlsUnfurlReq>) => {
+    const { url, documentId } = ctx.input.body;
+    const { user } = ctx.state.auth;
+    const { transaction } = ctx.state;
+    const urlObj = new URL(url);
+    const locale = dateLocale(user.language);
+    if (urlObj.protocol === "mention:") {
+      const { modelId: userId } = parseMentionUrl(url);
+
+      const [mentionedUser, document] = await Promise.all([
+        User.findByPk(userId, { transaction }),
+        Document.findByPk(documentId!, {
+          userId,
+          transaction,
+        }),
+      ]);
+      if (!mentionedUser) {
+        throw NotFoundError("Mentioned user does not exist");
+      }
+      if (!document) {
+        throw NotFoundError("Document does not exist");
+      }
+      authorize(user, "read", mentionedUser);
+      authorize(user, "read", document);
+
+      const lastView = head(orderBy(document.views, ["updatedAt"], ["desc"]));
+      const lastViewedAt = lastView ? lastView.updatedAt : undefined;
+      const lastActiveAt = mentionedUser.lastActiveAt;
+      let description;
+      if (lastViewedAt && differenceInMinutes(new Date(), lastViewedAt) < 5) {
+        description = t("Currently viewing", { ...opts(user) });
+      } else if (
+        lastActiveAt &&
+        differenceInMinutes(new Date(), lastActiveAt) < 5
+      ) {
+        description = lastViewedAt
+          ? t("Online now • Viewed {{ lastViewedAt }}", {
```

Comment:
Please split all these translations into two strings so that the • is not included and "Online now" doesn't need to be translated multiple times